app/src/pages/homes.tsx

```diff
@@ -98,30 +111,43 @@ const HomesPage: NextPage = () => {
         }
 
         const form = (event.target as HTMLFormElement);
-        const permissionsInputs = form.elements["Permissions"].entries();
+        const permissionsInputs = form.Permissions.entries();
         const permissions: Permission[] = []
         for(const [_, input] of permissionsInputs){
             if(input.checked){
                 permissions.push(input.value);
             }
         }
-       await editPermissions.mutateAsync({
-            user: form.elements["User"].value as string,
+       const permissionCheck = await editPermissions.mutateAsync({
+            user: form.User.value as string,
             homeId: selectedHome,
             permissions,
         });
-        setEditPermissionsModalOpen(false);
+        if(permissionCheck === "bad") {
+            setModalError("You cannot change your own Permissions!");
+        } else {
+            setEditPermissionsModalOpen(false);
+        }
     }
 
     const handleInvite = async (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
         const form = (event.target as HTMLFormElement);
         if(!selectedHome) return;
-        await inviteRoommate.mutateAsync({
+        if (!/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(form.email.value)) {
```

Comment:
I would use zod to validate this or their regex, they use a more extensive one to handle odd cases of emails

Reply:
I added this check in the latest commit.